`softshrink`: port to structured

Freey0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58050 Port silu to structured
* **#57623 Port softshrink to structured**
* #57622 Port hardsigmoid to structured
* #57621 Port leaky_relu to structured
* #57620 Port softplus to structured
* #57619 Port elu to structured

Differential Revision: [D28224703](https://our.internmc.facebook.com/intern/diff/D28224703)

Freey0 committed:
Port softshrink to structured

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28224703).

Freey0 committed:
Update on "Port softshrink to structured"


Differential Revision: [D28224703](https://our.internmc.facebook.com/intern/diff/D28224703)

[ghstack-poisoned]

Freey0 committed:
Update on "Port softshrink to structured"


Differential Revision: [D28224703](https://our.internmc.facebook.com/intern/diff/D28224703)

[ghstack-poisoned]

Freey0 committed:
Update on "Port softshrink to structured"


Differential Revision: [D28224703](https://our.internmc.facebook.com/intern/diff/D28224703)

[ghstack-poisoned]

Freey0 committed:
Update on "Port softshrink to structured"


Differential Revision: [D28224703](https://our.internmc.facebook.com/intern/diff/D28224703)

[ghstack-poisoned]

Freey0 committed:
Update on "Port softshrink to structured"


Differential Revision: [D28224703](https://our.internmc.facebook.com/intern/diff/D28224703)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28224703).

Freey0 committed:
Update on "Port softshrink to structured"


Differential Revision: [D28224703](https://our.internmc.facebook.com/intern/diff/D28224703)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28224703).
